File: app/src/types/constructor.ts
export interface PriceHistory {
  price: number
  changed_at: string
  changed_by: string
}

export interface Constructor {
  constructor_id: string
  name: string
  full_name: string
  nationality: string
  price: number
  status: 'active' | 'inactive'
  price_history: PriceHistory[]
  created_at: string
  updated_at: string
}

export interface ConstructorCreate {
  name: string
  full_name: string
  nationality: string
  price: number
  status?: string
}

export interface ConstructorUpdate {
  name?: string
  full_name?: string
  nationality?: string
  price?: number
  status?: string
}

File: service/app/schemas/constructor.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime

class PriceHistory(BaseModel):
    price: float
    changed_at: datetime
    changed_by: str = "system"

class ConstructorBase(BaseModel):
    name: str
    full_name: str
    nationality: str
    price: float
    status: str = "active"

class ConstructorCreate(ConstructorBase):
    pass

class ConstructorUpdate(BaseModel):
    name: Optional[str] = None
    full_name: Optional[str] = None
    nationality: Optional[str] = None
    price: Optional[float] = None
    status: Optional[str] = None

class ConstructorResponse(ConstructorBase):
    constructor_id: str
    price_history: List[PriceHistory] = []
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
